debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotWithSelectOverridesIT.java

```diff
@@ -99,10 +99,10 @@ public void takeSnapshotWithOverrides() throws Exception {
                         "dbo.table1,dbo.table3")
                 .with(
                         RelationalDatabaseConnectorConfig.SNAPSHOT_SELECT_STATEMENT_OVERRIDES_BY_TABLE + ".dbo.table1",
-                        "SELECT * FROM [dbo].[table1] where soft_deleted = 0 order by id desc")
+                        "SELECT * FROM [" + TestHelper.TEST_DATABASE + "].[dbo].[table1] where soft_deleted = 0 order by id desc")
```

Comment:
This is a breaking change. Being able to override snapshot `SELECT` statements is part of the public API but the existing statements won't work if they don't contain fully qualified table names because the connection no longer uses any database.

We need to pause here and introduce the configuration for opting into the breaking changes.

Reply:
@gunnarmorling what do you think? As for the opt-in configuration, we can introduce `database.names` which will accept a comma-separated list of database names (al planned) but will be limited to only one database at runtime until we implement iteration in the coordinator.

If the user specifies `database.names` instead of `database.dbname`, the connector will be allowed to expose all the breaking API changes (logging, metrics, topic naming, etc). Internally, it will translate to a flag that the components will use to switch the implementations similarly to `internal.implementation=legacy`.

Reply:
Yes, that's my thinking exactly: if `database.dbname` is given, everything should behave as of today, i.e. no need to qualify the DBs in filters or or statement overrides. Only when using the new option which takes a list of names, the breaking change of requiring the DB name in these places would apply.

Reply:
This means, that change to the test should not be needed.

Reply:
I refactored the test to cover both the single-partition and multi-partition configurations.